hash_practice/exercises.py

```diff
@@ -2,18 +2,51 @@
 def grouped_anagrams(strings):
     """ This method will return an array of arrays.
         Each subarray will have strings which are anagrams of each other
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(n)
     """
-    pass
+    anagrams_map = {}
+    if not strings:
+        return []
+    
+    for string in strings: 
+        letters_in_string = ''.join(sorted(string))
+        if letters_in_string in anagrams_map:
+            anagrams_map[letters_in_string].append(string)
+        else:
+            anagrams_map[letters_in_string] = [string]
+    
+    anagrams = []
+    for item in anagrams_map.values():
+        anagrams.append(item)
+    
+    return anagrams
+
+
 
 def top_k_frequent_elements(nums, k):
     """ This method will return the k most common elements
         In the case of a tie it will select the first occuring element.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)

```

Comment:
The call to `max` on line 43 will take O(m) time where `m` is the number of different frequency counts of the list (so we can put an upper bound of that at O(n)). `max` will be called up to `k` times, so the time complexity is actually O(k*n).